ubcpi/answer_pool.py

```diff
@@ -0,0 +1,240 @@
+        del existing[student_id_to_remove]
+    existing[student_id] = {}
+    pool[answer] = existing
+
+
+def offer_random(pool, answer, rationale, student_id):
+    offer_simple(pool, answer, rationale, student_id)
+
+
+def validate_seeded_answers_simple(answers, options, algo):
+    """
+    This validator checks if the answers includes all possible options
+    :param answers: the answers to be checked
+    :param options: all options that should exist in the answers
+    :return: None if everything is good. Otherwise, the missing option error message.
+    """
+    seen_options = {}
+    for answer in answers:
+        if answer:
+            key = options[answer['answer']].get('text')
+            if options[answer['answer']].get('image_url'):
+                key += options[answer['answer']].get('image_url')
+            seen_options.setdefault(key, 0)
+            seen_options[key] += 1
+
+    missing_options = []
+    index = 1
+    for option in options:
+        key = option.get('text') + option.get('image_url') if option.get('image_url') else option.get('text')
+        if seen_options.get(key, 0) == 0:
+            missing_options.append('Option ' + str(index))
+        index += 1
+
+    if missing_options:
+        return {'seed_error': 'Missing option seed(s): ' + ', '.join(missing_options)}
+
+    return None
+
+
+def validate_seeded_answers_random(answers):
+    """
+    This validator checks if there are a minimum of one answer
+    :param answers: the answers to be checked
+    :return: None if everything is good. Otherwise, the missing option error messag.
+    """
+    if len(answers) < 1:
+        return {'seed_error': 'Missing 1 option seed'}
+
+    return None
+
+
+def validate_seeded_answers(answers, options, algo):
+    """
+
+    :param answers: list of dict that contain seeded answers
+    :param algo:
+    :return: none if successful, otherwise error message
+    """
+    if algo['name'] == 'simple':
+        return validate_seeded_answers_simple(answers, options, algo)
+    elif algo['name'] == 'random':
+        return validate_seeded_answers_random(answers)
+    else:
+        raise UnknownChooseAnswerAlgorithm()
+
+
+def get_other_answers(answers, seeded_answers, get_student_item_dict, algo, options):
+    # "#" means the number of responses returned should be the same as the number of options.
+    num_responses = len(options) \
+        if 'num_responses' not in algo or algo['num_responses'] == "#" \
+        else int(algo['num_responses'])
+
+    if algo['name'] == 'simple':
+        return get_other_answers_simple(answers, seeded_answers, get_student_item_dict, num_responses)
+    elif algo['name'] == 'random':
+        return get_other_answers_random(answers, seeded_answers, get_student_item_dict, num_responses)
+    else:
+        raise UnknownChooseAnswerAlgorithm()
+
+
+def get_other_answers_simple(answers, seeded_answers, get_student_item_dict, num_responses):
```

Comment:
offer_simple is actually used to insert current student answer into the pool. And this is get other student's answers from the pool for current student.